packages/passport-interface/src/api/apiResult.ts

```diff
@@ -39,3 +39,12 @@ export type ResultMapper<T extends APIResult<unknown, unknown>> = {
   onValue: GetResultValue<T>;
   onError: GetError<T>;
 };
+
+/**
+ * Convenience type for errors which have a reason interpreted by the API
+ * layer, as well as error text.
+ */
+export interface ErrorWithReason<TReason> {
```

Comment:
if you wanted you could make this type more convenient to consume by setting a default type parameter, not a suggestion really just more of neat thing to do

```suggestion
export interface ErrorWithReason<TReason = string> {
```